apps/web/src/app/_components/header.tsx

```diff
@@ -21,6 +23,31 @@ interface HeaderProps {
  */
 export default function Header({ session, auth }: HeaderProps) {
   const pathname = usePathname();
+  const [showButton, setShowButton] = useState(true);
+
+  const profile = api.profile.getCurrentUser.useQuery();
+  const [profileId, setProfileId] = useState<string | undefined>(
+    profile.isSuccess ? profile.data?.id : undefined,
+  );
+
+  const reviews = api.review.getByProfile.useQuery(
+    { id: profileId! },
+    {
+      enabled: !!profileId,
+    },
+  );
+
+  useEffect(() => {
+    if (profile.isSuccess && profile.data) {
+      setProfileId(profile.data.id);
+    }
+  }, [profile]);
+
+  useEffect(() => {
+    if (reviews.isSuccess) {
+      setShowButton(reviews.data.length < 5);
+    }
+  }, [reviews]);
```

Comment:
Once we remove the `useState`s we can also remove the `useEffect`s and do something like the following:
```suggestion
  const showButton = (reviews.data?.length ?? 0) < 5;
```

Reply:
Yes, often it's the case you don't even need a useEffect since they cause a lot of unncessary rerenders most of the time.

I reread [this page](https://react.dev/learn/you-might-not-need-an-effect) so often it has good tips like the one rishi just recommended